recOrder/cli/compute_transfer_function.py

```diff
@@ -115,6 +60,96 @@ def compute_transfer_function_cli(config_path, output_path):
         f"Recreate this transfer function with:\n>> recorder compute-transfer-function {config_path} -o {output_path}"
     )
 
+def generate_and_save_birefringence_transfer_function(settings, dataset):
+    """Generates and saves the birefringence transfer function to the dataset, based on the settings.
+
+    Parameters
+    ----------
+    settings: object
+        Instance of the TransferFunctionSettings class.
+    dataset: object
+        NGFF Node to be updated.
+    """
+    echo_headline(
+        "Generating birefringence transfer function with settings:"
+    )
+    echo_settings(settings.birefringence_transfer_function_settings)
+
+    # Calculate transfer functions
+    intensity_to_stokes_matrix = (
+        inplane_oriented_thick_pol3d.calculate_transfer_function(
+            **settings.birefringence_transfer_function_settings.dict()
+        )
+    )
+    # Save
+
+    dataset[
+        "intensity_to_stokes_matrix"
+    ] = intensity_to_stokes_matrix.cpu().numpy()[None, None, None, ...]
+
+def generate_and_save_phase_transfer_function(settings, dataset):
+    """Generates and saves the phase transfer function to the dataset, based on the settings.
+
+    Parameters
+    ----------
+    settings: object
```

Comment:
Annotating non-builtin types with `object` is an unfortunate legacy this code base inherited from Java developers. In Python we should use the actual type in both type hints and docstrings. In this case `TransferFunctionSettings`.

Reply:
See an example here:
https://github.com/mehta-lab/recOrder/blob/6f8f05f500c9298053caeeba6f35788b738317e4/recOrder/acq/acq_functions.py#L120-L143

Reply:
Sounds good. I'll make those changes. I noticed that they put the default arguments into the parameters while I have mine in the description/summary of compute_transfer_function_cli. Should I change that to match as well?

Reply:
In `compute_transfer_function_cli` you can't really move the default into the signature. The argument takes a str path, but the defualt (or actually fallback) behavior does not read the settings from any file. This is ok as long as it is documented.